sdk/core/System.Net.ClientModel/src/Pipeline/MessagePipeline.cs

```diff
@@ -95,7 +135,12 @@ public static MessagePipeline Create(PipelineOptions options)
     // TODO: note that without a common base type, nothing validates that MessagePipeline
     // and Azure.Core.HttpPipeline have the same API shape. This is something a human
     // must keep track of if we wanted to add a common base class later.
-    public ClientMessage CreateMessage() => _transport.CreateMessage();
+    public ClientMessage CreateMessage(RequestOptions options)
```

Comment:
@KrzysztofCwalina, I changed this and want to make sure to get your eyes on it.  

Are we happy requiring RequestOptions to be passed to CreateMessage?  
The reason I added this was so we could make RequestOptions.Apply protected rather than public.

Reply:
Actually, I decided against this for several reasons:
- I want to make the CreateMessage factory method as simple as possible
  - Passing RequestOptions in and calling apply on it there feels hidden and a little convoluted and error-prone
- Making RequestOptions.Apply public and requiring client-authors to call it lets us pass options from the client-author into that method and keep them separate from client-user provided options that are properties on RequestOptions.  This mirrors the pattern we're using in ClientPipeline.GetPipeline and makes it easier to implement precedence rules cleanly